### PR TITLE
[docs] add some ML specific terminology

### PR DESCRIPTION
**Pull request by compnerd**
*Description*:

These come up quite commonly in conversation.  Add a handful of them to the
lexicon.

*Testing*:
n/a

*Documentation*:
This PR is to add doc.

### CODE REVIEW

File: docs/Lexicon.md [hunk 1]
```diff
@@ -2,16 +2,58 @@
...
   Do Not Merge
 
+* DNN
+  Deep Neural Networks.  A neural network with multiple layers between the input
+  and output layers.  These work well with linear and non-linear relationships.
+
+### G
+
+* GEMM
+  Generialied Matrix Multiply.
```
Comment (rdzhabarov) [inline docs/Lexicon.md]: nit: General Matrix Multiply

File: docs/Lexicon.md [hunk 2]
```diff
@@ -20,3 +62,15 @@
 * NFCI
    No Functional Change Intended
```
Comment (rdzhabarov) [inline docs/Lexicon.md]: what's the difference between NFC and NFCI :)? What are the situation you'd use NFCI?

File: docs/Lexicon.md [hunk 3]
```diff
@@ -2,16 +2,58 @@
 
 ### Definitions
 
+### A
+
+* ANN
+  Artificial Neural Network.  A computational framework based on a synthetic
```
Comment (rdzhabarov) [inline docs/Lexicon.md]: do you intend to render this on a new line. It's not currently, check the "view" button.

File: docs/Lexicon.md [hunk 4]
```diff
@@ -2,16 +2,58 @@
...
+
+* GRU
+  Gated Recurrent Unit.  A gating mechanism for neural networks.  They are
+  similar to LSTM but exhibit better performance characteristics on smaller data
+  sets.
+
+### L
+
+* LSTM
+  Long Short Term Memory.  Units of RNNs consisting of a cell, input gate,
```
Comment (rdzhabarov) [inline docs/Lexicon.md]: there are quite a few double spacing here and there in the file.

**Comment by SplitInfinity** [comment]
Consider adding the following:
*Collaboration Lingo*
- WIP (work in progress)
- LGTM (looks good to me)

*Compiler Lingo*
- JIT (just-in-time compile)
- AOT (ahead-of-time compile)
- BC (bytecode)

As an overall note, I know that most of the contributors are compiler experts and new to machine learning and that's why the lexicon is mostly full of machine learning terminology. But consider adding some frequently used compiler jargon as well for those of us (like myself) who are more familiar with the ML side of things than the compiler side of things.

File: docs/Lexicon.md (hunk 2 above)
Comment (compnerd) [reply docs/Lexicon.md]: I suppose, *technically* when you can prove mathematically that there is no functional change.  Mostly it is when its clearly an algebraic change versus what you expect to have no functional side-effects.  Most people infer that it was the intention if it turns out not to be the case.

**Comment by SplitInfinity** [comment]
@compnerd 

I added IR to my list. These four are the ones I've heard most often since I joined the team.

File: docs/Lexicon.md (hunk 4 above)
Comment (bertmaher) [reply docs/Lexicon.md]: 2spaces4life

**Pull request merged**